lib/registrant.rb

```diff
@@ -0,0 +1,54 @@
+require './lib/vehicle'
+
+class Registrant
+    attr_reader :name, 
+                :age, 
+                :permit,
+                :license_data,
+                :vehicles,
+                :year
+    attr_accessor :balance
+    def initialize(name, age, permit = false)
+        @name = name 
+        @age = age
+        @permit = permit
+        @year = year
+        @license_data = {:written=>false, :license=>false, :renewed=>false}
+        @vehicles = []
+        @balance = 0
+        @antique = antique?
+        @ev_vehicle = electric_vehicle?
+    end
+
+    def permit?
+        @permit 
+    end
+    
+    
+    def earn_permit
+        @permit = true
+    end
+
+    def add_vehicle(vehicle)
+        @vehicles << vehicle
+    end
+    def balance
+      @balance = 0    
+         if @vehicle.antique?
+          25
+        if @vehicle.electric_vehicle?
+        200
+        else
+       100
+        end
+    end
```

Comment:
I don't think this method is in the interaction pattern, so it does not need to be implemented. The `Facility` should keep track of its own `collected_fees`, but the `Registrant` does not have a balance.